src/test/java/com/contrastsecurity/maven/plugin/it/stub/FakeContrastAPI.java

```diff
@@ -221,7 +276,30 @@ private static void uploadCodeArtifact(final HttpExchange exchange) {
 
   private static void startScan(final HttpExchange exchange) {
     discardRequest(exchange);
-    final String json = "{\"id\": \"scan-id\"}";
+    final String json =
+        "{\n"
+            + "  \"id\": \""
+            + SCAN_ID
+            + "\",\n"
+            + "  \"organizationId\": \""
+            + ORGANIZATION_ID
+            + "\",\n"
+            + "  \"projectId\": \""
+            + PROJECT_ID
+            + "\",\n"
+            + "  \"codeArtifactId\": \""
+            + CODE_ARTIFACT_ID
+            + "\",\n"
+            + "  \"status\": \"WAITING\",\n"
+            + "  \"createdTime\": \"2021-07-26T18:45:19.222+00:00\",\n"
+            + "  \"startedTime\": null,\n"
+            + "  \"completedTime\": null,\n"
+            + "  \"language\": \"JAVA\",\n"
+            + "  \"label\": \"scan.http\",\n"
+            + "  \"errorMessage\": null,\n"
+            + "  \"includeNamespaceFilters\": [],\n"
+            + "  \"excludeNamespaceFilters\": []\n"
+            + "}";
```

Comment:
I found it helpful to paste JSON verbatim from the API, because then I didn't have a chance to screw anything up trying to tranlsate JSON to Java ;-). I think that was fine for writing the tests initially, but I agree that it's not readable. 

I don't think a `Map` would be more readable, but we can use the `Scan`, `Project`, etc types for most of these. The problem is that some the types are incomplete, so we wouldn't be testing with JSON that is exactly as it's returned from the API unless we filled out those models. In some cases I was just lazy and didn't want to put all the fields I wasn't going to use in the types, but I think I've since corrected that. There's one exception in a branch that I have not yet opened a PR for. The paged results that the Scan API returns are pretty obtuse. I'm 90% certain the API is returning a Spring Data type that was never meant to be exposed to REST clients. I'd rather spend my time lobbying to improve the API than model what we have currently (it's a lot fields).

What do you think about extracting these String literals to JSON files in `src/test/resources`? That would avoid any problems manually translating from JSON to Java, and it would side-step the incomplete model issue I described. The trade-off I see is that everything would no longer be in one file - you'd have to open up each JSON file to see its contents.

Thoughts?

Reply:
Extracting into a file sounds like a nice solution.

Java multiline string might have been a nice solution for this 😭 Maybe in the future we could compile tests against Java 15 🤷

Reply:
Yeah... I have been waiting for JetBrains to fix this forever https://youtrack.jetbrains.com/issue/IDEA-85478?_ga=2.153457200.94513435.1627566585-1898392706.1627566585

I extracted the JSON to a file and this file is easier to read and has less duplication